react/LocationSearch.tsx

```diff
@@ -1,37 +1,59 @@
-import React, { useState, useMemo } from 'react'
+import React, { useState, useMemo, useRef } from 'react'
 import { FormattedMessage } from 'react-intl'
-import { Input, IconSearch, IconClear } from 'vtex.styleguide'
+import { Input, IconSearch, IconClear, IconWarning } from 'vtex.styleguide'
+import { positionMatchWidth } from '@reach/popover'
+
 import {
   Combobox,
   ComboboxInput,
   ComboboxOption,
   ComboboxPopover,
   ComboboxList,
-} from '@reach/combobox'
-import '@reach/combobox/styles.css'
+} from './components/Combobox'
+import PlaceIcon from './components/PlaceIcon'
 
-import { addresses as mockedAddresses } from './addresses'
-import styles from './LocationSearch.css'
+interface Interval {
+  offset: number
+  size: number
+}
 
-const MAX_DROPDOWN_ADDRESSES = 6
+export interface Suggestion {
+  description: string
+  mainText: string
+  mainTextMatchInterval: Interval
+  secondaryText: string
+}
 
-// This function will be replaced in the future, after integrating the
-// component with GraphQL queries.
-const getAddresses = (searchTerm: string) => {
-  return mockedAddresses
-    .filter((address: string) =>
-      address.toLocaleLowerCase().includes(searchTerm.toLocaleLowerCase())
-    )
-    .slice(0, MAX_DROPDOWN_ADDRESSES)
+const renderSuggestionText = (suggestion: Suggestion) => {
+  const { mainText } = suggestion
+  const { offset, size } = suggestion.mainTextMatchInterval
+  return (
+    <div className="truncate c-muted-2">
+      <span className="c-on-base">{mainText.substr(0, offset)}</span>
+      <span className="c-on-base b">{mainText.substr(offset, size)}</span>
+      <span className="c-on-base">{mainText.substr(size + offset)}</span>
```

Comment:
can we opt in to a simpler markup? like the following
```suggestion
      <span className="c-on-base">
        {mainText.substr(0, offset)}
        <span className="b">{mainText.substr(offset, size)}</span>
        {mainText.substr(size + offset)}
      </span>
```

Reply:
I think so, it looks better indeed. I was just worried that this wouldn't be good at a semantic level, but thinking again, I guess there is nothing wrong with a nested `span`.

Reply:
yep, it's an inline element inside another inline element, so I think we should be fine